File: 1608-Special-Array-With-X-Elements-Greater-Than-or-Equal-X/main.go
package main

func specialArray(nums []int) int {

    sort.Slice(nums, func(i, j int) bool {
        return nums[i] < nums[j]
    })

    n := len(nums)

    for i := 1; i < n + 1; i++{
        count := 0
        for _, num := range nums {
            if num >= i{
                count++
            }
        }
        if count == i {
            return i
        }
    }

    return -1
}

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/main.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        
        nums.sort()
        n = len(nums)

        for x in range(1, n+1):
            count = 0
            for num in nums:
                if num >= x:
                    count += 1
            if count == x:
                return x
        
        return -1
